### ISSUE TITLE
indentation_width is not working well with multiple line if and guard

### ISSUE THREAD

**Issue by miku1958** [realm/SwiftLint#4961]
### New Issue Checklist

- [ ] Updated SwiftLint to the latest version
- [ ] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

For example:
![image](https://user-images.githubusercontent.com/24806909/236386070-6c17e4e6-7394-4098-ab55-bec3fc4149c3.png)

##### Complete output when running SwiftLint, including the stack trace and command used

Swift package plugin

### Environment

* SwiftLint version: 0.51.0
* Installation method used: SPM
* Paste your configuration file:

```yml
opt_in_rules:
  - indentation_width

indentation_width:
   severity: warning 
   indentation_width: 4
   include_comments: false
   include_compiler_directives: false
   trueinclude_multiline_strings: false
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)?
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
        guard let startSubvalue = subvalue(at: gesture.startLocation, minutes: minutes),
              let currentSubvalue = subvalue(at: gesture.location, minutes: minutes) else {
            return nil
        }
```

**Comment by shmilyQin** [comment]
Facing the same issue. could you tell me whether you fix it? Thanks.

**Comment by SimplyDanny** [comment]
I suppose this is not straightforward to fix. I'm not going to invest energy into this in the near future. But everyone is free to propose a PR. 😉

**Comment by tanaev** [comment]
> I suppose this is not straightforward to fix. I'm not going to invest energy into this in the near future. But everyone is free to propose a PR. 😉

Please take a look
